cloverage/test/cloverage/instrument_test.clj

```diff
@@ -217,10 +241,63 @@
           (let [[[message-type _ message]] @log-messages]
             (t/is (= :error
                      message-type))
-            (doseq [s ["Error instrumenting form"
+            (doseq [s ["Error evaluating instrumented form"
                        "Unable to resolve symbol: this-function-does-not-exist"]]
               (t/is (str/includes? message s)
                     (str "Error message should include %s" (pr-str s))))))
         (t/testing "We should have attempted to evauluate the *original* form"
           (t/is (= true
                    @evalled-original-form?)))))))
+
+(t/deftest instrument-java-interop-forms-test
+  (t/testing "Java interop forms should get instrumented correctly (#304)"
+    (t/testing "Method calls"
+      ;; these two syntaxes are equivalent
+      (t/testing "(. object method & args) syntax"
+        (t/is (= '(do (. clojure.lang.RT count (do [(do 3) (do 4)])))
+                 (rw/macroexpand-all (inst/instrument-form #'inst/nop nil '(. clojure.lang.RT count [3 4]))))))
+      (t/testing "(. object (method & args)) syntax"
+        (t/is (= '(do (. clojure.lang.RT (count (do [(do 3) (do 4)]))))
+                 (rw/macroexpand-all (inst/instrument-form #'inst/nop nil '(. clojure.lang.RT (count [3 4]))))))))))
```

Comment:
This is a test for #305 which I think actually only was an issue in this PR. There were previously no tests for the Java interop forms here so I thought it was a good idea to add some